src/system-tests/tests/suites/rest-api/ssm-targets.ts

```diff
@@ -0,0 +1,47 @@
+import { SsmTargetHttpService } from '../../../../http-services/targets/ssm/ssm-target.http-services';
+import { DigitalOceanSSMTarget } from '../../../digital-ocean/digital-ocean-ssm-target.service.types';
+import { configService, logger, testTargets } from '../../system-test';
+import { ssmTestTargetsToRun } from '../../targets-to-run';
+
+export const ssmTargetRestApiSuite = () => {
+    describe('SSM Target REST API Suite', () => {
+        let ssmTargetService: SsmTargetHttpService;
+
+        beforeAll(() => {
+            ssmTargetService = new SsmTargetHttpService(configService, logger);
+        });
+
+        test('5742: Get an SSM target by ID', async () => {
+            const doTarget = testTargets.get(ssmTestTargetsToRun[0]) as DigitalOceanSSMTarget;
+            const ssmTarget = await ssmTargetService.GetSsmTarget(doTarget.ssmTarget.id);
+            expect(ssmTarget.id).toEqual(doTarget.ssmTarget.id);
+        }, 15 * 1000);
+
+        test('5743: Get all SSM targets', async () => {
+            const ssmTargets = await ssmTargetService.ListSsmTargets(false);
+            // verify that each of the expected SSM targets is in the returned list
+            ssmTestTargetsToRun.forEach(testTarget => {
+                const doTarget = testTargets.get(testTarget) as DigitalOceanSSMTarget;
+                expect(ssmTargets).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining({id: doTarget.ssmTarget.id})
+                    ]));
+            });
```

Comment:
Is it possible to do this in a single `expect` call instead of 1 per `ssmTestTargetsToRun`? Something like this...haven't tried it though:


```suggestion
            expect(ssmTargets).toEqual(
                expect.arrayContaining(
                    ssmTestTargetsToRun.map(
                        testTarget => expect.objectContaining({ id: (testTargets.get(testTarget) as DigitalOceanSSMTarget).ssmTarget.id }
            ))));
```

Reply:
Interesting idea. I tried to think of a way to do a single call but didn't come up with this one. I'll try it with a test run that includes multiple targets.

Reply:
I tested to make sure it fails as expected, too.

```bash
● SSM Target REST API Suite › 5743: Get all SSM targets

    expect(received).toEqual(expected) // deep equality

    Expected: ArrayContaining [ObjectContaining {"id": "ddd"}, ObjectContaining {"id": "ddd"}]
    Received: [{"agentId": "BDUE/ZNUUMBC43p/Irhr5BA0GTHGQB0tze74SjZ/7u0=", "agentPublicKey": "rvqLqcuPr+C/F1esNYwi31XHhWq2g3M2eY4tx7PG1bw=", "agentVersion": "3.0.732.32", "allowedTargetUsers": [], "allowedVerbs": [], "environmentId": "91faab50-b20e-4b7c-a4a2-5746f8a2a8ba", "id": "a043b1de-97cb-4f3e-a4f9-e2446f318a32", "name": "st-1djn2h9fzsnsu4p-al2-pm-IvNyGBv51sOqfW0", "region": "us-east-1", "status": "Online", "timeLastStatusUpdate": "2022-05-05T20:46:56.737214+00:00"}, {"agentId": "MFVUeMddG+La1pfOU9q8WdbGYnz4tyGL/vKoS4jXFnI=", "agentPublicKey": "8VO4hhqQiGlGbfBbs67/NXxb5R3F+ZOigpDc7SvXzns=", "agentVersion": "3.0.732.32", "allowedTargetUsers": [], "allowedVerbs": [], "environmentId": "91faab50-b20e-4b7c-a4a2-5746f8a2a8ba", "id": "e982a3fa-edf8-4e98-9d9f-59da2831a9a6", "name": "st-1djn2h9fzsnsu4p-debian11-pm-SnYt92jOcy1lmI3", "region": "us-east-1", "status": "Online", "timeLastStatusUpdate": "2022-05-05T20:44:56.739979+00:00"}]
```

Reply:
bb2495bcadf1207fef8030241d7c355d76693d3f